src/Logging/Logging.Batching/BatchLogService.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CodeMonkeys.Logging.Batching
+{
+    public abstract class BatchLogService<TOptions> : ScopedLogService<TOptions>
+        where TOptions : BatchLogOptions, new()
+    {
+        private BlockingCollection<LogMessage> _queue;
+
+        public override bool IsEnabled
+        {
+            get => base.IsEnabled;
+            protected set
+            {
+                if (value)
+                    Run();
+
+                base.IsEnabled = value;
+            }
+        }
+
+        protected BatchLogService(string context) 
+            : base(context)
+        {
+            Run();
+        }
+
+        protected abstract Task PublishMessageBatch(IEnumerable<LogMessage> messageBatch);
+
+        protected override void PublishMessage(LogMessage message)
+        {
+            if (_queue.IsAddingCompleted)
+                return;
+
+            try
+            {
+                _queue.Add(message);
+            }
+            catch { }
+        }
+
+        private void Run()
+        {
+            _queue = Options.QueueCapacity == null ?
+                new BlockingCollection<LogMessage>(new ConcurrentQueue<LogMessage>()) :
+                new BlockingCollection<LogMessage>(
+                    new ConcurrentQueue<LogMessage>(),
+                    Options.QueueCapacity.Value);
+
+            Task.Run(ProcessingLoop);
+        }
+
+        private async Task ProcessingLoop()
+        {
+            var currentBatch = new List<LogMessage>(Options.BatchCapacity ?? int.MaxValue);
+
+            while (IsEnabled)
+            {
+                TakeBatch(currentBatch);
+
+                if (currentBatch.Count > 0)
+                    await PublishMessageBatch(currentBatch);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5626772)